tests/Resources/ti.ui.textfield.addontest.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2011-Present by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Ti, Titanium */
+/* eslint no-unused-expressions: "off" */
+'use strict';
+var should = require('./utilities/assertions');
+
+describe('Titanium.UI.TextField', function () {
+	var win;
+
+	afterEach(function () {
+		if (win) {
+			win.close();
+		}
+		win = null;
+	});
+
+	it.ios('hasText', function (finish) {
+		win = Ti.UI.createWindow();
+		var textFieldA = Ti.UI.createTextField({
+			top: '60dip',
+			value: 0
+		});
+
+		win.add(textFieldA);
+
+		var textFieldB = Ti.UI.createTextField({
+			top: '120dip',
+			value: 0
+		});
+
+		win.add(textFieldB);
+
+		textFieldA.addEventListener('change', function () {
```

Comment:
What's the change event listener for? How will it be triggered in a unit test?

Reply:
When value get set , at that time also 'change' event listener get called.

Reply:
This event will only be triggered by user input, not when changing the text programmatically.

Reply:
In this particular case when we set number, 'change' event listener is getting called. I guess you are correct, ideally it should not call if it set programatically.
Though I have updated unit test, I'll create a ticket to check the behaviour of 'change' event listener.

Reply:
For verifying the conversion of 0 to "0" -
There is no getter method implemented in this to get value from textfield. Whenever we call getter it will simply fetch value from proxy (default getter). So with getter it will always give the same value, which is number. Other places also I can see default getter (value from proxy) returned.